### DISCUSSION TITLE
無法顯示 SimpleTweaksPlugin 的分解差值

### DISCUSSION THREAD

**Discussion by qwer20108** [GpointChen/FFXIVChnTextPatch-GP#320] [Q&A] [answered]
```
1361,"精製魔晶石：<hex:020812E1E80201FF04>可<hex:FF07>不可<hex:03>　武具投影：<hex:020812E1E80301FF04>可<hex:FF07>不可<hex:03>　<hex:02083CE1E80401FF1F>建議分解技能：<hex:022003E80503>.00<hex:FF16>道具分解：不可<hex:03>"

1361,マテリア精製:<hex:02080EE1E80201FF04>○<hex:FF03>×<hex:03>　武具投影:<hex:02080EE1E80301FF04>○<hex:FF03>×<hex:03>　<hex:020831E1E80401FF20>分解適正スキル:<hex:022003E80503>.00<hex:FF0A>分解:×<hex:03>
```

可能原因 中文分隔號 ：是全形分號 但是英文日文為一般分號 :
所以我 trace SimpleTweaksPlugin code 發現這行好像會比對 
```$": {item.LevelItem.Row},00" ```
才進入下一步.
https://github.com/Caraxi/SimpleTweaksPlugin/blob/main/Tweaks/Tooltips/DesynthesisSkill.cs#L81

我手動改 Addon.csv 把 ：replace 成 : 則變成 ui 無法顯示分解等級.

請問有人知道為什麼單純更改字串 ui 會錯誤嗎?

如果 replace 成 日文或是英文則 ui 顯示正常

**Comment by GpointChen** [comment, marked as answer]
你想把全形冒號換成半形冒號的話需要修改與之相關的指令碼（e.g. `<hex:02083CE1E80401FF1F>`）

例如上面這行中`3C`這個byte代表了接下來整段指令碼的長度，而`1F`這個byte則代表了其中「`建議分解技能：<hex:022003E80503>.00`」這段的長度

全形冒號是3個byte，半形冒號是1個byte，所以換成半形冒號後代表指令碼會減少2個byte，故3C應改成3A，1F應改成1D

更多關於指令碼的處理可以參考[這篇筆記](https://hackmd.io/@GpointChen/SJi_gv-ad#29-%E6%8C%87%E4%BB%A4%E7%A2%BC%E8%99%95%E7%90%86)。
